### ISSUE TITLE
how to pair a sklearn-regressor model to pymoo optimization algorithm?

### ISSUE THREAD

**Issue by osvaldoaesthetic** [msu-coinlab/pymoo#83]
I want to implement  a multi-objective optimization algorithm in pymoo, however i dont have an explicit model, in this case my objective function is a random forest from scikit learn.

does anyone of you know how to set the problem to the out[“F”] ?

I have..
____________________________________________________________

`import` numpy as np
`from` sklearn.ensemble import RandomForestRegressor

np.matrix...
`pred=x[:]`
target=y[:]

`forest=RandomForestRegressor(n_estimators=50,` max_depth=None, oob_score=True)
`trainedf=forest.fit(pred,target)`

__________________________________________________________-
then

import numpy as np
from pymoo.model.problem import Problem

`class` MyProblem(Problem):

    def __init__(self):
        super().__init__(n_var=3,
                         n_obj=3,
                        
                         xl=np.array([25,10,10]),
                         xu=np.array([100,60,25]))

    def _evaluate(self, x, out, *args, **kwargs):
        
        f1 = trainedf
      
        out["F"] = ([f1])
   
__________________________________________________________________
the randomforest runs OK, but the problem is probably here:

  def _evaluate(self, x, out, *args, **kwargs):
        
        f1 = trainedf
      
        out["F"] = ([f1])

any other option please.

**Comment by Peng-YM** [comment]
Actually `pymoo` does not care about what your model is. What you need to do is implement your own `_evaluate` function. You have an input matrix (or vector), then you can use an external model to calculate the fitness value `F`. Then just put it to `out` using `out['F'] = F`.

**Comment by julesy89** [comment]
Yes Peng-YM is right. Your problem description is a bit vague to derive what you are intending to do. Mostly I have seen pymoo being used to optimize hyperparameters of machine learning models.
That means you are building your model in the `_evaluate` from given the parameters `x`. And then you define your objectives, often based on accuracy and complexity.

**Comment by osvaldoaesthetic** [comment]
> Yes Peng-YM is right. You have to set your objectives to `out['F']`.
> Your problem description is a bit vague to derive what you are intending to do. Mostly, in Machine Learning I have seen pymoo being used to optimize hyperparameters of models.
> That means you are building your model in the `_evaluate` from given the parameters `x`. And then you define your objectives, often based on accuracy and complexity.

**Comment by osvaldoaesthetic** [comment]
Thank you @julesy89 yes,I did not explain what im pretending to do at all. 

I mean, I have a problem with 3 variables **(x1,x2,x3)** and 3 responses **(y1, y2, y3)** , this were modeled by a randomforest, and the RF's hyperparemeters were already optimized.

Now, I want to find the combinations of Xs(x1,x2,x3) **''pareto front''** that optimize the responses **Y's(y1,y2,y3)**

**Comment by julesy89** [comment]
This is frequently done for engineering problems especially when the evaluation itself is quite expensive. You RF functions as a so called Response Surface. 
You like to use the RF to predict the `(y1,y2,y3)` which means you have to assign the prediction of the RF to your `out['F']`.
